### PR TITLE
Updating solver convergence checks

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
In PR #627 it was pointed out that our current approach for checking for solver convergence is somewhat specific to IPOPT, and may not work for other solvers. This PR updates our convergence checks to use Pyomo's `check_optimal_termination` method.

## Changes proposed in this PR:
- Replace all checks for `TerminationCondition.optimal` (and most `SolverStatus.ok`) with `check_optimal_termination`.
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by eslickj — APPROVED** [review]
Looks good.  Really cleans things up.

File: idaes/generic_models/unit_models/column_models/solvent_condenser.py [hunk 1]
```diff
@@ -27,11 +27,10 @@
 
 # Import Pyomo libraries
 from pyomo.environ import (
+    check_optimal_termination,
```
Comment (jsiirola) [inline idaes/generic_models/unit_models/column_models/solvent_condenser.py:30]: Is this import necessary, given the only call to it is commented out?

File: idaes/generic_models/unit_models/column_models/tray_column.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 from pyomo.common.config import ConfigBlock, ConfigValue, In, Bool
 from pyomo.network import Arc, Port
 from pyomo.environ import value, Integers, RangeSet, TransformationFactory, \
-    Block, Reference, SolverStatus, TerminationCondition
+    Block, Reference, check_optimal_termination
```
Comment (jsiirola) [inline idaes/generic_models/unit_models/column_models/tray_column.py:25]: Is this import necessary given the only use is commented out?

File: idaes/generic_models/unit_models/column_models/solvent_condenser.py (hunk 1 above)
Comment (andrewlee94) [reply idaes/generic_models/unit_models/column_models/solvent_condenser.py:30]: Strictly no, but that test really should be fixed and uncommented, so I left the import in place.

**Review by jsiirola — APPROVED** [review]

**Pull request merged**